### PR TITLE
Add Minicart Component

### PR DESCRIPTION
**Pull request by vitoria**
#### What is the purpose of this pull request?
Add the first version of mini cart component.

#### What problem is this solving?
Show the items that the user added to the cart.

#### Screenshots or example usage
https://minicart--storecomponents.myvtex.com/_v/minicart
captura de tela de 2018-04-25 16-49-54

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: pages/pages.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "pages": {
+    "minicart": {
+      "path": "/",
+      "theme": "theme"
```
Comment (estacioneto) [inline pages/pages.json]: delet this. Import the CSS in the component you need

File: react/theme.css [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    height: 300px;
+}
+
+.arrow-up {
+    width: 0; 
+    height: 0; 
+    border-left: 10px solid transparent;
+    border-right: 10px solid transparent;
+    border-bottom: 10px solid white;
+  }
```
Comment (estacioneto) [inline react/theme.css]: Put the component css separate (minicart.css, for example) and import it in the component file

File: react/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import MiniCartButton from './MiniCartButton'
+
+export default function HelloWorld() {
+  return (
+    <div className="relative w-100 bg-red h6">
+      <div className="absolute right-0 mr6"><MiniCartButton labelButtonFinishShopping={'Finalizar Compra'} labelMiniCartEmpty={'Sua sacola está vazia!'} /></div>
+    </div>
```
Comment (estacioneto) [inline react/index.js]: Are you using lint? :thinking:

File: react/MiniCart.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      <div className="minicart-item shadow-3 pa4 flex items-center justify-center">
+        <Spinner />
+      </div>
+    )
+  }
+
+  render() {
+    const { data, labelMiniCartEmpty, labelButton } = this.props
+    let content
+    console.log(data)
```
Comment (estacioneto) [inline react/MiniCart.js]: hm

File: manifest.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  "registries": [
+    "smartcheckout"
+  ],
+  "settingsSchema": {},
+  "scripts": {
+    "postreleasy": "vtex publish --public"
+  },
+  "dependencies": {
+    "vtex.store-graphql": "2.x"
+  }
```
Comment (salesfelipe) [inline manifest.json]: I think that the policies are missing

File: react/MiniCart.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+import Spinner from '@vtex/styleguide/lib/Spinner'
+
+/**
+ * Minicart component
+ */
+class MiniCart extends Component {
+  handleClickButton = () => location.assign('/checkout/#/cart')
+
+  renderWithoutProducts = (label) => {
+    return (
```
Comment (estacioneto) [inline react/MiniCart.js]: You return directly in `handleClickButton`... Why not here?

**Review by estacioneto — DISMISSED** [review]

File: react/MiniCartButton.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ import PropTypes from 'prop-types'
 import Button from '@vtex/styleguide/lib/Button'
 import CartIcon from './CartIcon'
 import MiniCart from './MiniCart'
+import './global.css'
```
Comment (estacioneto) [inline react/MiniCartButton.js]: Import all css in all components? Divide it :)

File: react/MiniCart.js (hunk 4 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/index.js (hunk 3 above)
Comment (vitoriaHeliane) [reply react/index.js]: Yes!

File: react/MiniCart.js [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+      content = this.renderWithoutProducts(labelMiniCartEmpty)
+    } else {
+      content = this.renderMiniCartWithItems(data.orderForm, labelButton)
+    }
+    return content
+  }
+}
+
+MiniCart.propTypes = {
+  /* Informations about order form */
```
Comment (salesfelipe) [inline react/MiniCart.js]: Shouldn´t this be the products on the cart?

File: react/MiniCartItem.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+MiniCartItem.propTypes = {
+  /* Minicart item's image */
```
Comment (salesfelipe) [inline react/MiniCartItem.js]: Better to suppress the 'Minicart', like: Item's image...

File: react/MiniCartItem.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  imageUrl: PropTypes.string.isRequired,
+  /* Minicart item's name */
```
Comment (salesfelipe) [inline react/MiniCartItem.js]: idem

File: react/MiniCartItem.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  name: PropTypes.string.isRequired,
+  /* Minicart item's selling price */
```
Comment (salesfelipe) [inline react/MiniCartItem.js]: idem

File: react/MiniCartItem.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  sellingPrice: PropTypes.number.isRequired,
+  /* Minicart item's list price */
```
Comment (salesfelipe) [inline react/MiniCartItem.js]: idem

File: react/MiniCartItem.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+  listPrice: PropTypes.number.isRequired,
+  /* Minicart item's sku */
```
Comment (salesfelipe) [inline react/MiniCartItem.js]: idem

File: react/global.css [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+}
+.top-100 {
+    top: 100%
+}
+
+.top-50 {
+    top: 50%
+}
+
+.minicart-item {
```
Comment (salesfelipe) [inline react/global.css]: You should follow the pattern defined by @brunojdo 

vtex-nomedaapp-componente__subcomponente

Like: 

vtex-shelf__arrow-left

File: react/package.json [hunk 15]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "vtex.render-getting-started",
```
Comment (salesfelipe) [inline react/package.json]: Name of the app

**Review by salesfelipe — CHANGES_REQUESTED** [review]

File: react/MiniCart.js (hunk 6 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/package.json (hunk 15 above)
Comment (vitoria) [reply react/package.json]: Done!

File: react/MiniCartItem.js (hunk 13 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCartItem.js (hunk 12 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCartItem.js (hunk 11 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCartItem.js (hunk 10 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCartItem.js (hunk 9 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCart.js (hunk 8 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+import './global.css'
+
+/**
+ * Minicart component
+ */
+class MiniCart extends Component {
+  handleClickButton = () => location.assign('/checkout/#/cart')
+
+  renderWithoutProducts = (label) => (
+    <div className="minicart-item pa4 shadow-4 flex items-center justify-center">
```
Comment (claudivanfilho) [inline react/MiniCart.js]: `minicart-item` -> `vtex-minicart__item`

File: react/MiniCart.js [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+      <span className="f5">{label}</span>
+    </div>
+  )
+
+  renderMiniCartWithItems = (orderForm, label) => {
+    let contentStyle = 'pa4'
+    if (orderForm.items.length > 3) {
+      contentStyle += ' overflow-auto minicart-content mb4'
```
Comment (claudivanfilho) [inline react/MiniCart.js]: `contentStyle` -> `contentClass`
`minicart-content` -> `vtex-minicart__content` e também não precisa desse `if`

Faz assim:

```
let contentStyle = 'vtex-minicart__content overflow-auto pa4 mb4'
```
daí no css troca isso `height: 300px;` por `max-height: 300px;`

Dessa maneira a classe `vtex-minicart__content` sempre vai estar presente no escopo e não só quando o num de items for maior que 3

File: react/MiniCart.js [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+    }
+    return (
+      <div className="flex flex-column">
+        <div className="arrow-up self-end mr3 pr1"></div>
```
Comment (claudivanfilho) [inline react/MiniCart.js]: `arrow-up` ->  `vtex-minicart__arrow-up`

File: react/MiniCart.js [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+        <div className="shadow-3">
+          <div className={contentStyle}>
+            {orderForm.items.map(item => (
+              <div className="flex flex-row" key={item.id}>
+                <MiniCartItem
```
Comment (claudivanfilho) [inline react/MiniCart.js]: ``` <MiniCartItem {...item} /> ```

File: react/MiniCart.js [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+                showListPrice={false} />
+            </div>
+          </div>
+        </div>
+      </div >
+    )
+  }
+
+  renderLoading = () => (
+    <div className="minicart-item shadow-3 pa4 flex items-center justify-center">
```
Comment (claudivanfilho) [inline react/MiniCart.js]: `vtex-minicart__item`

File: react/MiniCartItem.js [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+import './global.css'
+
+/**
+ * Minicart item component
+ */
+export default class MiniCartItem extends Component {
+  render() {
+    const { imageUrl, name, skuName, sellingPrice, listPrice } = this.props
+    return (
+      <div className="minicart-item flex flex-row relative bb b--silver mb3">
```
Comment (claudivanfilho) [inline react/MiniCartItem.js]: vtex-minicart__item

File: react/MiniCartItem.js [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+        <img className="image-size" src={imageUrl} />
+        <div className="ml3">
+          <div className="name-size mt3">
```
Comment (claudivanfilho) [inline react/MiniCartItem.js]: `name-size` -> `vtex-minicart__item-name`

File: react/MiniCartItem.js [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+            <span className="b">{name}</span>
+          </div>
+          <div>
+            <div className="sku-size">
```
Comment (claudivanfilho) [inline react/MiniCartItem.js]: `sku-size` -> `vtex-minicart__sku-name`

File: react/global.css [hunk 24]
```diff
@@ -0,0 +1,54 @@
+body {
+    width: 100%;
+}
+.top-100 {
+    top: 100%
+}
+
+.top-50 {
```
Comment (claudivanfilho) [inline react/global.css]: essa class nunca está sendo usada

File: react/global.css [hunk 25]
```diff
@@ -0,0 +1,54 @@
+body {
+    width: 100%;
+}
+.top-100 {
```
Comment (claudivanfilho) [inline react/global.css]: deve existir uma alternativa melhor do que esse `top: 100%`

**Review by claudivanfilho — CHANGES_REQUESTED** [review]

File: react/global.css (hunk 25 above)
Comment (vitoria) [reply react/global.css]: Eu não conheço outra. Você tem alguma sugestão?

File: react/global.css (hunk 24 above)
Comment (vitoria) [reply react/global.css]: Done!

File: react/MiniCartItem.js (hunk 23 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCartItem.js (hunk 22 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCartItem.js (hunk 21 above)
Comment (vitoria) [reply react/MiniCartItem.js]: Done!

File: react/MiniCart.js (hunk 20 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js (hunk 18 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js (hunk 17 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js (hunk 16 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js (hunk 19 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/global.css (hunk 14 above)
Comment (vitoria) [reply react/global.css]: Done!

File: manifest.json (hunk 5 above)
Comment (vitoria) [reply manifest.json]: Done!

File: react/MiniCart.js (hunk 17 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js [hunk 26]
```diff
@@ -0,0 +1,106 @@
...
+import Spinner from '@vtex/styleguide/lib/Spinner'
+import './global.css'
+
+/**
+ * Minicart component
+ */
+class MiniCart extends Component {
+  handleClickButton = () => location.assign('/checkout/#/cart')
+
+  renderWithoutProducts = (label) => (
```
Comment (andreldsa) [inline react/MiniCart.js]: Products or Items

File: react/MiniCart.js [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+    <div className="vtex-minicart__item pa4 shadow-4 flex items-center justify-center">
+      <span className="f5">{label}</span>
+    </div>
+  )
+
+  renderMiniCartWithItems = (orderForm, label) => (
+    <div className="flex flex-column">
+      <div className="vtex-minicart__arrow-up self-end mr3 pr1"></div>
+      <div className="shadow-3">
+        <div className="pa4 overflow-auto vtex-minicart__content">
```
Comment (andreldsa) [inline react/MiniCart.js]: Should be the first class `vtex-minicart__content`

File: react/MiniCart.js [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+  render() {
+    const { data, labelMiniCartEmpty, labelButton } = this.props
+    let content
+    if (data.loading) {
+      content = this.renderLoading()
+    } else if (data.orderForm.length) {
+      content = this.renderWithoutProducts(labelMiniCartEmpty)
+    } else {
+      const orderForm = {
+        items: [
```
Comment (andreldsa) [inline react/MiniCart.js]: mock?

File: react/index.js [hunk 29]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import MiniCartButton from './MiniCartButton'
+import './global.css'
+
+export default function HelloWorld() {
```
Comment (andreldsa) [inline react/index.js]: Don't need this.

**Review by andreldsa — CHANGES_REQUESTED** [review]

File: react/MiniCart.js (hunk 27 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCart.js (hunk 28 above)
Comment (vitoria) [reply react/MiniCart.js]: DOne!

File: react/index.js (hunk 29 above)
Comment (vitoria) [reply react/index.js]: Done!

File: react/MiniCart.js (hunk 26 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

**Review by salesfelipe — APPROVED** [review]

File: react/package.json [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+  "name": "minicart",
+  "version": "0.0.0",
+  "scripts": {
+    "pretest": "yarn",
+    "test": "jest"
+  },
+  "dependencies": {
+    "@vtex/product-details": "^1.1.0",
+    "@vtex/styleguide": "2.0.0-rc.39",
+    "lodash": "^4.17.5"
```
Comment (estacioneto) [inline react/package.json]: Need this dependency?

**Review by estacioneto — DISMISSED** [review]

File: react/package.json (hunk 30 above)
Comment (vitoria) [reply react/package.json]: No, I'm going to remove it.

File: react/MiniCart.js [hunk 31]
```diff
@@ -0,0 +1,84 @@
...
+    </div>
+  )
+
+  render() {
+    const { data, labelMiniCartEmpty, labelButton } = this.props
+    let content
+    if (data.loading) {
+      content = this.renderLoading()
+    } else if (!data.orderForm.length) {
+      console.log(labelMiniCartEmpty)
```
Comment (andreldsa) [inline react/MiniCart.js]: Remove

**Review by andreldsa — APPROVED** [review]
Fix then LGTM.

**Review by claudivanfilho — APPROVED** [review]

File: react/CartIcon.js [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+  /* Fill color for the icon */
+  fillColor: PropTypes.string,
+}
+
+CartIcon.defaultProps = {
+  size: 20,
+  fillColor: '#000000',
+}
+
+export default CartIcon
```
Comment (brunojdo) [inline react/CartIcon.js]: put this directly in the class.

File: react/MiniCart.js [hunk 33]
```diff
@@ -0,0 +1,83 @@
...
+      content = this.renderWithoutItems(labelMiniCartEmpty || this.context.intl.formatMessage({ id: 'minicart-empty' }))
+    } else {
+      content = this.renderMiniCartWithItems(data.orderForm,
+        labelButton || this.context.intl.formatMessage({ id: 'finish-shopping-button-label' }))
+    }
+    return content
+  }
+}
+
+MiniCart.propTypes = {
```
Comment (brunojdo) [inline react/MiniCart.js]: put this into the classe. Like carousel

File: react/MiniCartButton.js [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+              labelMiniCartEmpty={labelMiniCartEmpty}
+              labelButton={labelButtonFinishShopping} />
+          </div>
+        }
+      </div>
+    )
+  }
+}
+
+MiniCartButton.propTypes = {
```
Comment (brunojdo) [inline react/MiniCartButton.js]: same here

**Review by brunojdo — CHANGES_REQUESTED** [review]
Where is the locale intl?
Where's the changelog?

File: react/CartIcon.js (hunk 32 above)
Comment (vitoria) [reply react/CartIcon.js]: Done!

File: react/MiniCart.js (hunk 33 above)
Comment (vitoria) [reply react/MiniCart.js]: Done!

File: react/MiniCartButton.js (hunk 34 above)
Comment (vitoria) [reply react/MiniCartButton.js]: Done!

**Review by brunojdo — APPROVED** [review]

File: pages/pages.json [hunk 35]
```diff
@@ -0,0 +1,12 @@
+{
+  "pages": {
+    "minicart": {
+      "path": "/"
```
Comment (brunojdo) [inline pages/pages.json]: `"path":  "/_v/minicart"`

**Review by brunojdo — CHANGES_REQUESTED** [review]

File: react/setupTests.js [hunk 36]
```diff
@@ -0,0 +1,10 @@
+import Enzyme from 'enzyme'
```
Comment (brunojdo) [inline react/setupTests.js]: destructure this` { configure } from 'enzyme'`

File: react/package.json [hunk 37]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "vtex.minicart",
+  "version": "0.0.1",
```
Comment (brunojdo) [inline react/package.json:3]: remove

**Review by brunojdo — APPROVED** [review]

**Pull request merged**